64jam.p8

```diff
@@ -126,82 +131,81 @@ function update_projectiles()
 	end
 end
 
-function fire_rocket()
-	if (player.rockets > 0) then
-		local x = 0
-		local y = 0
+-- converts a direction and speed to a dx, dy
+function dir_to_deltas(dir, speed)
+	local dx = 0
+	local dy = 0
+
+	if dir == 12 or dir == 10 or dir == 1 then
+		dy = -speed
+	end
+	
+	if dir == 5 or dir == 6 or dir == 7 then
+		dy = speed
+	end
+	 
+	if dir >= 1 and dir < 6 then
+		dx = speed
+	end
+	if dir > 6 and dir <= 10 then
+		dx = -speed
+	end 
+	return dx, dy
```

Comment:
Fantastic!  I didn't know you could return like this.  I initially wrote two functions (figure out dx, figure out dy) but I wasn't happy with it so rolled back for the time being.  Good stuff.